pkg/services/internal/whatsonchain/broadcast.go

```diff
@@ -26,60 +25,114 @@ type broadcastRequest struct {
 			return StatusAlreadyBroadcasted, txid, nil
-		case strings.Contains(responseText, "txn-mempool-conflict"):
+		case containsI(responseText, "txn-mempool-conflict"):
 			return StatusDoubleSpend, txid, nil
-		case strings.Contains(responseText, "missing inputs"):
+		case containsI(responseText, "Missing inputs"):
 			return StatusMissingInputs, txid, nil
 		default:
-			return StatusError, "", fmt.Errorf("woc returned error %d: %s", res.StatusCode(), responseText)
+			return StatusError, "", fmt.Errorf("woc returned unexpected error %d: %s", res.StatusCode(), responseText)
 		}
 	}
 
-	if txResp.Txid != txid {
-		return StatusError, txResp.Txid, fmt.Errorf("woc returned txid %s does not match local calculated txid %s", txResp.Txid, txid)
+	if resp.TxID != txid {
+		return StatusError, "", fmt.Errorf("txid mismatch: expected %s, got %s", txid, resp.TxID)
 	}
 
 	return StatusSuccess, txid, nil
 }
 
-func convertNotes(notes []string) wdk.Notes {
-	converted := make(wdk.Notes, len(notes))
-	for i, note := range notes {
-		now := time.Now()
-		converted[i] = wdk.ReqHistoryNote{
-			When: &now,
-			What: note,
-			Args: nil,
+func (woc *WhatsOnChain) fetchTxInfo(ctx context.Context, txid string) (*txInfoResponse, error) {
+	var info txInfoResponse
+
+	req := woc.httpClient.
+		R().
+		SetContext(ctx).
+		SetResult(&info).
+		AddRetryCondition(retryOnTooManyRequestsStatus)
+
+	url := fmt.Sprintf("%s/tx/%s/info", woc.url, txid)
+	res, err := req.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch tx info: %w", err)
+	}
+
+	if res.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("failed to retrieve tx info: status %d, body: %s", res.StatusCode(), res.String())
+	}
+
+	return &info, nil
+}
+
+func (woc *WhatsOnChain) processSingleTx(ctx context.Context, txid string, rawTx []byte) wdk.PostedTxID {
+	status, returnedTxid, broadcastErr := woc.broadcast(ctx, rawTx)
+
+	if broadcastErr != nil {
+		return wdk.PostedTxID{
+			Result: wdk.PostedTxIDResultError,
+			TxID:   txid,
+			Error:  broadcastErr,
+			Notes:  convertNotes([]string{fmt.Sprintf("broadcast error: %v", broadcastErr)}),
 		}
 	}
-	return converted
+
+	resultStatus, notes := classifyBroadcastStatus(status)
+
+	blockHash, blockHeight, fetchErr := woc.tryFetchTxInfo(ctx, returnedTxid, &notes)
+
+	return wdk.PostedTxID{
+		Result:       resultStatus,
+		TxID:         returnedTxid,
+		DoubleSpend:  status == StatusDoubleSpend || status == StatusMissingInputs,
+		AlreadyKnown: status == StatusAlreadyBroadcasted,
+		BlockHash:    blockHash,
+		BlockHeight:  blockHeight,
+		Error:        firstNonNilError(fetchErr),
+		Notes:        convertNotes(notes),
+	}
+}
+
+func (woc *WhatsOnChain) tryFetchTxInfo(ctx context.Context, txid string, notes *[]string) (string, int64, error) {
+	info, err := woc.fetchTxInfo(ctx, txid)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to fetch tx info: %v", err)
+		*notes = append(*notes, errMsg)
+		return "", 0, fmt.Errorf(errMsg)
+	}
```

Comment:
Warn: printf: non-constant format string in call to fmt.Errorf
Besides, no error wrapping. 

I'm suggesting sth like this:
```suggestion
	if err != nil {
		*notes = append(*notes, fmt.Sprintf("failed to fetch tx info: %v", err))
		return "", 0, fmt.Errorf("failed to fetch tx info: %w", err)
	}
```

Reply:
Already addressed 👍 
Resolving... :)